pg_shard.c

```diff
@@ -713,25 +768,123 @@ ClassifyRestrictions(List *queryRestrictList, List **remoteRestrictList,
 
 
 /*
- * RowAndColumnFilterQuery builds a query which contains the filter clauses from
+ * ClassifyHasAggregates decides whether local and distributed queries will include
+ * aggregates or not. remoteQueryHasAggregates and localQueryHasAggregates are output
+ * parameters to receive these boolean values.
+ */
+static void
+ClassifyHasAggregates(Query *query, bool *remoteQueryHasAggregates,
+					  bool *localQueryHasAggregates)
+{
+	*remoteQueryHasAggregates = false;
+	*localQueryHasAggregates = false;
+
+	if (query->groupClause != NIL)
+	{
+		Oid tableId = ExtractFirstDistributedTableId(query);
+		bool safeToPushDownAggregate = SafeToPushDownAggregate(query, tableId);
+
+		/* if we can push down the aggregates update */
```

Comment:
This comment doesn't make any sense to me. I'd just remove it: the code is clear as-is.